packages/app/src/background/services/credentials/index.ts

```diff
@@ -0,0 +1,205 @@
+import browser from "webextension-polyfill";
+
+import CryptoService, { ECryptMode } from "@src/background/services/crypto";
+import HistoryService from "@src/background/services/history";
+import NotificationService from "@src/background/services/notification";
+import SimpleStorage from "@src/background/services/storage";
+import { OperationType, VerifiableCredential } from "@src/types";
+import { parseVerifiableCredentialFromJson } from "@src/util/credential";
+
+import type { IBackupable } from "@src/background/services/backup";
+
+const VERIFIABLE_CREDENTIALS_KEY = "@@VERIFIABLE-CREDENTIALS@@";
+
+export default class VerifiableCredentialsService implements IBackupable {
+  private static INSTANCE: VerifiableCredentialsService;
+
+  private verifiableCredentialsStore: SimpleStorage;
+
+  private cryptoService: CryptoService;
+
+  private historyService: HistoryService;
+
+  private notificationService: NotificationService;
+
+  private constructor() {
+    this.verifiableCredentialsStore = new SimpleStorage(VERIFIABLE_CREDENTIALS_KEY);
+    this.cryptoService = CryptoService.getInstance();
+    this.historyService = HistoryService.getInstance();
+    this.notificationService = NotificationService.getInstance();
+  }
+
+  static getInstance(): VerifiableCredentialsService {
+    if (!VerifiableCredentialsService.INSTANCE) {
+      VerifiableCredentialsService.INSTANCE = new VerifiableCredentialsService();
+    }
+
+    return VerifiableCredentialsService.INSTANCE;
+  }
+
+  addVerifiableCredential = async (verifiableCredentialJson: string): Promise<boolean> => {
+    if (!verifiableCredentialJson) {
+      return false;
+    }
+
+    const verifiableCredential = await parseVerifiableCredentialFromJson(verifiableCredentialJson);
+
+    if (verifiableCredential === null) {
```

Comment:
This condition wont be needed it we throw errors in `validateVerifiableCredential()` function.